src/login/msal/cachePlugin.ts

```diff
@@ -0,0 +1,37 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ICachePlugin, TokenCacheContext } from '@azure/msal-node';
+import { credentialsSection } from '../../constants';
+import { KeyTar, tryGetKeyTar } from '../../utils/keytar';
+import { getSelectedEnvironment } from '../environments';
+
+const keytar: KeyTar | undefined = tryGetKeyTar();
+
+const beforeCacheAccess = async (cacheContext: TokenCacheContext): Promise<void> => {
+	if (keytar) {
+		try {
+            const cachedValue: string | null = await keytar.getPassword(credentialsSection, (await getSelectedEnvironment()).name);
+			cachedValue && cacheContext.tokenCache.deserialize(cachedValue);
+		} catch (error) {
+			console.log(JSON.stringify(error));
```

Comment:
I'm not a big fan of `console.log` in production code - it should either be in the output channel or not logged at all. Was this just here for debugging purposes? In any case, I don't think we want to log any errors for keytar/password-related stuff

Reply:
I think that was an artifact of the sample code. Just removed it